src/Tizen.Network.Stc/Tizen.Network.Stc/StcRule.cs

```diff
@@ -0,0 +1,288 @@
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            if(disposing)
+            {
+                // destroy managed resources
+            }
+
+            int ret = Interop.Stc.Rule.Destroy(_ruleHandle);
+            if (ret == (int)StcError.None)
+            {
+                _ruleHandle = IntPtr.Zero;
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Failed to destroy Rule handle, Error - " + (StcError)ret);
+            }
+
+            _disposed = true;
+        }
+
+        /// <summary>
+        /// A property for Application ID for statistics rule.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <value>AppId.</value>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown while setting this property when Stc is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown while setting this value due to an invalid operation.</exception>
+        public string AppId
+        {
+            get
+            {
+                string appId;
+                int ret = Interop.Stc.Rule.GetAppId(_ruleHandle, out appId);
+                if (ret != (int)StcError.None)
+                {
+                    Log.Error(Globals.LogTag, "Failed to get AppId, Error - " + (StcError)ret);
+                    return string.Empty;
+                }
+                return appId;
+            }
+            set
+            {
+                if (_disposed)
+                {
+                    throw new ObjectDisposedException("Invalid StcRule instance (Object may have been disposed or released)");
+                }
+                int ret = Interop.Stc.Rule.SetAppId(_ruleHandle, value);
+                if (ret != (int)StcError.None)
+                {
+                    Log.Error(Globals.LogTag, "Failed to set AppId, Error - " + (StcError)ret);
+                    StcErrorFactory.ThrowStcException(ret);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Set the time interval for statistics rule.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <param name="from">The beginning of the time interval.</param>
+        /// <param name="to">The end of the time interval.</param>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown when the Stc is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the method failed due to an invalid operation.</exception>
+        public void SetTimeInterval(DateTime from, DateTime to)
```

Comment:
Why need it? you also provide `From` `To` property

Reply:
(capi)SetTimeInterval(_ruleHandle, from, to): It needs both "from and to" parameters. So one approach i find suitable is to use a method as setter. The "From and To" properties are getters only.

Reply:
Why do you provide a getter of `From`?
It is a value passed from user

Reply:
Yes, 'From' is set by user. But Core package provide getter API too, and hence i also replicated it. I was told to replicate the functionalities provided by Core STC package. As Core package recently had changes and may have more changes in future, so i thought such APIs may get used in next changes.

Reply:
PLEASE, forget Core API, It is not Core API. We can't accept low usability API.
Please, thinks about API usability